Revit_Core_Engine/Query/Intersections.cs

```diff
@@ -34,6 +34,29 @@ public static partial class Query
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Intersection result between curve and solid. For no intersection, an empty list is returned.")]
+        [Input("curve", "Curve to get the intersection for.")]
+        [Input("solid", "Solid to get the intersection for.")]
+        [Output("curves", "List of intersecting curves.")]
+        public static List<Curve> Intersections(this Curve curve, Solid solid)
+        {
+            if (curve == null || solid == null)
+            {
+                return null;
+            }
+
+            List<Curve> curves = new List<Curve>();
+            SolidCurveIntersection intersection = solid.IntersectWithCurve(curve, new SolidCurveIntersectionOptions());
```

Comment:
Sanity check: does this method never return null? I remember that it can happen for projections/inersections. So could be good to add a null check for `intersection`, just in case.

Reply:
Agreed 👍